### PR TITLE
[FLOSS H2] Core: Added else error to kdbc

### PR DESCRIPTION
**Pull request by 4ydan**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by lawli3t** [comment]
LGTM! :rocket:

**Review by flo91 — CHANGES_REQUESTED** [review]
Thank you for the PR!
The change looks good and all buildserver checks passed. :+1:
I think it would be interesting to briefly describe why the other changes were reverted in the commit a8a3340ddef4be510a08b1ef35a4c9deba4f3919.

I also noticed some small points for improvement:

- According to https://www.libelektra.org/tutorials/contributing-with-clion#commiting-your-changes, the commit messages should have the format _module: short statement_

The module is not mentioned in your commit messages.


If the checklist for PRs is followed, two points of the category _Basics_ need attention:
- **Reference to issue**
	Most of the PRs for the FLOSS H2 task don't have references to issues.
	I'm not really sure if this point applies to our PRs.
	Should we create new issues for the changes we made?

- **The PR is rebased with current master.**
In https://www.libelektra.org/devgettingstarted/contributing#creating-a-pull-request is stated in point 6: _keep in mind to sync your fork - fetch, rebase & push_
When I open the repository for the PR (https://github.com/4ydan/libelektra/tree/kdbc_error_check), there is the message _This branch is 4 commits ahead, 11 commits behind ElektraInitiative:master._ So it looks like this has still to be done.

A really great addition would be an unit test that fails on the old version and succeeds on the version with the changes of this PR! :scroll:

File: src/libs/elektra/kdb.c
Comment (markus2330) [inline src/libs/elektra/kdb.c:681]: Please also call ELEKTRA_SET_ERROR with some information that there is no resolver plugin or kdbGet was missing in the resolver plugin.

**Comment by markus2330** [comment]
Amazing job, please add "ready to merge" once it is ready to merge.

**Comment by markus2330** [comment]
Please rebase the PR.

>  Shall we create new issues for the changes we made?

In H3 yes, in H2 it is not important.

**Comment by markus2330** [comment]
Thank you! Can you add tests within T2?

**Pull request merged**
